From 5c2d8d2d9da6423eec076fd51049d7b4677b61c6 Mon Sep 17 00:00:00 2001
From: Tao Liu <ltao@redhat.com>
Date: Tue, 17 Aug 2021 16:21:43 +0800
Subject: [PATCH 24/27] Set gdb max-value-size to be unlimited

gdb-10.2 uses max-value-size as the maximum size in bytes that the contents
of a object may allocate. The default value of max-value-size is 64K. However,
it could be not enough for allocating an object which requires larger space, and
failed at the startup of crash.

In gdb-7.6, there is no max-value-size check and works fine. So in this patch,
let's just set max-value-size to be unlimited.

Signed-off-by: Tao Liu <ltao@redhat.com>
---
 gdb_interface.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gdb_interface.c b/gdb_interface.c
index b098ed823d55..2fe9ccde83a1 100644
--- a/gdb_interface.c
+++ b/gdb_interface.c
@@ -300,6 +300,13 @@ retry:
 	sprintf(req->buf, "set width 0");
 	gdb_interface(req);
 
+#ifdef GDB_10_2
+	req->command = GNU_PASS_THROUGH;
+	req->name = NULL, req->flags = 0;
+	sprintf(req->buf, "set max-value-size unlimited");
+	gdb_interface(req);
+#endif
+
 #if 0
        /*
         *  Patch gdb's symbol values with the correct values from either
-- 
2.30.2

